themes/default/content/blog/redis-cloud-provider/index.md

```diff
@@ -0,0 +1,306 @@
+  tags: {
+    Name: "pulumi-redis-cloud-tester"
+  },
+  iamInstanceProfile: instanceProfile.name,
+  userData: `#!/bin/bash
+  sudo amazon-linux-extras install redis6
+  `
+});
+```
+
+And again, deploy your stack:
+
+```bash
+pulumi up
+```
+
+{{% notes type="info" %}}
+If you're new to Pulumi, you may notice that when you run the `pulumi up` command this time, Pulumi does not attempt to re-create the resources you've already declared. This is because Pulumi is a _declarative_ tool: You only need to specify the desired end state of your resources, and Pulumi will figure out what steps (if any) are necessary to get from your infrastructure's current state to your declared desired state.
+{{% /notes %}}
+
+## Testing the Architecture
+
+Now that you have your EC2 instance up and running, it's time to verify that you can connect to your Redis cluster and run commands:
+
+1. In the AWS Console, find the EC2 instance named `pulumi-redis-cloud-tester`, and connect to it via SSM Systems Manager:
+
+    ![AWS EC2 console, listing available instances](ec2-console.png)
+    ![AWS EC2 connect dialog](ec2-connect.png)
+
+1. In the [Redis Cloud console](https://app.redislabs.com/), locate your database, and select "Connect":
+
+1. In the connection dialog, copy the connection info for the Redis CLI:
+
+    ![Redis Cloud console showing the connection dialog with the CLI option selected](redis-console.png)
+
+1. Paste the command into the Systems Manager shell prompt, which will connect you to the Redis instance. From there you can run Redis commands like `INCR` that will write to the Redis cluster:
+
+    ![SSM Systems Manager shell with commands like redis connect and INCR](shell-commands.png)
+
+Now your workloads can use your Redis Cloud cluster! You can tear down all the resources you created with the following command:
+
+```bash
+pulumi destroy
+```
+
+## Next Steps and Conclusion
+
+Now that you have proven that the architecture works, here are some next steps you might want to take with this codebase:
+
+* Replace the EC2 test instance with a real workload that uses Redis:
+  * If your workload is running on EC2, then the code in this post should get you up and running. However, if you find you need some help [Pulumi AI](https://www.pulumi.com/ai) can be a very helpful resource. Simply enter a natural language query and select a support Pulumi language, and Pulumi AI will generate a Pulumi program that can be copy/pasted into your program and iterated on in the same Pulumi AI conversation.
+  * If you're using another AWS compute platform, Pulumi also has helpful guides for creating workloads on [ECS](https://www.pulumi.com/docs/clouds/aws/guides/ecs/), [EKS](https://www.pulumi.com/docs/clouds/aws/guides/eks/), or [Lambda](https://www.pulumi.com/docs/clouds/aws/guides/lambda/).
+* Try creating [Redis Cloud clusters peered Google Cloud](https://www.pulumi.com/registry/packages/rediscloud/api-docs/subscriptionpeering/#gcp).
```

Comment:
Is there a word missing here? Should it be "Redis Cloud clusters peered with Google Cloud", or something like that?

Reply:
You're correct. Fixed.